### DISCUSSION TITLE
GRDBQuery 0.9 with custom GRDB

### DISCUSSION THREAD

**Discussion by Jason-Abbott** [groue/GRDBQuery#53] [Q&A]
I am curious only for quick tips since I haven’t worked too hard at this yet.

The initial upgrade to 0.9 in an Xcode project with multiple targets, and using a custom GRDB build, brought back the familiar `Unable to find module dependency: ‘CSQLite’` for each target.

I can submodule GRDBQuery into my project or try to sort out the build config but was popping in here just to see if there was something quick to consider first.

Thanks!

**Comment by groue** [comment]
I'll let other people share their own experience. In the kind of apps I build (no custom GRDB build), I had some some success in avoiding this painful Xcode hiccup with `@_exported import GRDB`: https://github.com/groue/GRDB.swift/issues/1424#issuecomment-1774088155

**Reply by Jason-Abbott** [reply]
Thanks @groue . That thread gives me several things to verify and try.

I’ll get back to the branch where I was working on this in a bit and comment here with any learnings.

**Comment by Jason-Abbott** [comment]
Use of `@_exported import GRDB` in my target that uses GRDB didn't alleviate the problem. I also made a fork of GRDBQuery that I tried to make happy without an explicit GRDB dependency (like version <= 0.8).

Unsurprisingly, that broke in several places. Since I'm already using a git submodule for the custom GRDB build, I went ahead and did basically the same for GRDBQuery rather than keep wrestling for something fancier.

My usage may be a bit of an edge case but I'll detail my steps in case it's useful.

- Create a git submodule of GRDBQuery
- Add a local target also called `GRDBQuery` (so existing imports don't have to change)
- I updated the new target as follows (most is specific to my config):
   - Select "Convert to Group" on target folder so I can *reference* GRDBQuery files in place rather than copying them
   - Add `GRDBCustom` as a dependency (instead of GRDB)
   - Exclude the GRDBQuery submodule folder from `.swiftlint`
   - Disable `Existential any` upcoming feature
   - Disable `Default Internal Imports` upcoming feature
- Add the *submodule* `Sources/GRDBQuery` files to the `GRDBQuery` *target*

That is enough for my project to build and its targets still to refer to `GRDBQuery` without changes to those targets or changes to the GRDBQuery source files.

**Reply by groue** [reply]
Thanks for sharing your build setup 🙏

**Comment by groue** [comment]
What would you think, @Jason-Abbott, if GRDBQuery were embedded right into GRDB?

A few `#if canImport(SwiftUI)` and availability checks should do the trick... What do I miss?

**Reply by Jason-Abbott** [reply]
That makes sense to me! It would simplify _my_ usage.

I'm on holiday with just phone so won't be able to consider more thoroughly for a couple days.

**Reply by groue** [reply]
Take your time :) Could it bother someone? If GRDBQuery is embedded in GRDB7, the next major version, the breaking change would be to remove the dependency on GRDBQuery, and that's all 🤔
